src/main/java/uk/gov/hmcts/reform/civil/referencedata/LocationRefDataService.java

```diff
@@ -33,6 +33,56 @@ public class LocationRefDataService {
     private final LRDConfiguration lrdConfiguration;
     private final AuthTokenGenerator authTokenGenerator;
 
+    public LocationRefData getCtscLocationUnSpec(String authToken) {
+        try {
+            ResponseEntity<List<LocationRefData>> responseEntity = restTemplate.exchange(
+                buildURIforCtscUnspec(),
+                HttpMethod.GET,
+                getHeaders(authToken),
+                new ParameterizedTypeReference<List<LocationRefData>>() {
+                }
+            );
+            List<LocationRefData> ctscLocations = responseEntity.getBody();
+            if (ctscLocations == null || ctscLocations.isEmpty()) {
+                log.warn("Location Reference Data Lookup did not return any CTSC location for Salford CTSC");
+                return LocationRefData.builder().build();
+            } else {
+                if (ctscLocations.size() > 1) {
+                    log.warn("Location Reference Data Lookup returned more than one CTSC location for Salford CTSC");
+                }
+                return ctscLocations.get(0);
+            }
+        } catch (Exception e) {
+            log.error("Location Reference Data Lookup Failed for Salford CTSC - " + e.getMessage(), e);
+        }
+        return LocationRefData.builder().build();
```

Comment:
if lookup fails, is this valid to have empty Location ?

Reply:
I have rewritten the method written by GA as getCcmccLocation with new locations. We are not using them anywhere in our civil-service code so not sure why they want it like that.

Reply:
I think the common service should ideally throw some custom exception Not found etc. then based on the usage requirement user can ignore the error or rethrow this

Reply:
@deepthidoppalapudihmcts Please can you confirm for Sabah's comment if we are ok with empty location. Thank you

Reply:
@asthamalviya I agree with Sabah to throw Appropriate exception